ibm/service/database/validators.go

```diff
@@ -0,0 +1,146 @@
+// Copyright IBM Corp. 2017, 2022 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package database
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/IBM/cloud-databases-go-sdk/clouddatabasesv5"
+)
+
+/* TODO move other validators in here */
+
+/* VERSION VALIDATOR */
+
+type Version struct {
+	Version     string
+	Type        string
+	Status      string
+	IsPreferred bool
+	Transitions []VersionTransition
+}
+
+type VersionTransition struct {
+	Application         string
+	Method              string
+	SkipBackupSupported bool
+	FromVersion         string
+	ToVersion           string
+}
+type AllowedUpgrade struct {
+	ToVersion           string `json:"to_version"`
+	SkipBackupSupported bool   `json:"skip_backup_supported"`
+}
+
+func getAllowedUpgradeVersions(versions []Version) []AllowedUpgrade {
+	var allowedVersions []AllowedUpgrade
+
+	for _, version := range versions {
+		for _, transition := range version.Transitions {
+			if transition.Method == "restore" {
+				allowedVersions = append(allowedVersions, AllowedUpgrade{
+					ToVersion:           transition.ToVersion,
+					SkipBackupSupported: transition.SkipBackupSupported,
+				})
+			}
+		}
+	}
+
+	return allowedVersions
+}
+
+func (v *Version) isVersionAllowed(version string) bool {
+	for _, transition := range v.Transitions {
+		if transition.ToVersion == version {
+			return true
+		}
+	}
+	return false
+}
+
+func (v *Version) isSkipBackupAllowed(version string) bool {
+	for _, transition := range v.Transitions {
+		if transition.ToVersion == version {
+			return transition.SkipBackupSupported
+		}
+	}
+	return false
+}
+
+func expandVersions(versions []clouddatabasesv5.VersionsCapabilityItem) []*Version {
+	if len(versions) == 0 {
+		return nil
+	}
+
+	expandedVersions := make([]*Version, 0, len(versions))
+
+	for _, capabilityVersion := range versions {
+		version := Version{
+			Version:     *capabilityVersion.Version,
+			Status:      *capabilityVersion.Status,
+			IsPreferred: *capabilityVersion.IsPreferred,
+		}
+
+		if capabilityVersion.Transitions != nil {
+			var transitions []VersionTransition
+
+			for _, transition := range capabilityVersion.Transitions {
+				transitions = append(transitions, VersionTransition{
+					Application:         *transition.Application,
+					Method:              *transition.Method,
+					SkipBackupSupported: *transition.SkipBackupSupported,
+					FromVersion:         *transition.FromVersion,
+					ToVersion:           *transition.ToVersion,
+				})
+			}
+			version.Transitions = transitions
+		}
+		expandedVersions = append(expandedVersions, &version)
+	}
+
+	return expandedVersions
+}
+
+func validateVersion(instanceID string, targetVersion string, skipBackup bool, meta interface{}) (err error) {
+	// Get available versions for deployment
+	capability, err := getDeploymentCapability("versions", instanceID, "classic", "us-south", meta)
```

Comment:
This could be moved into a function `fetchVersions` with an optional deployment ID.

```go
func fetchVersions(instanceId string, meta interface{}) ([]*Version versions) {
	capability, err := getDeploymentCapability("versions", instanceID, "classic", "us-south", meta)
	// ... turn SDK Versions into TF Versions
	return versions
}
```

You could mock the CLI response with the expected API response and validate the versions returned match the expected output